### PR TITLE
Support uint64 literals

### LINKED ISSUE #6145: Support uint64 literals
### Is your feature request related to a problem or challenge?

It is not possible to provide numeric literals greater than int64 max without them being treated as either float (lossy) or decimal (not currently castable to uint64)

### Describe the solution you'd like

We should attempt to parse uint64s after int64 parsing fails, but before falling back to float or decimal

### Describe alternatives you've considered

I looked at using decimals, but this would require another change to make it possible to cast uint64 to decimal

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jackkleeman**
It should be possible to compare numeric types to uint64 literals. Currently for anything above the int64 max, it will become a float or a decimal128 depending on config. The float conversion is lossy, and the decimal type is currently not comparable with uint64, so both of these are problematic.

# Which issue does this PR close?

Closes #6145.

# Are these changes tested?

Yes

# Are there any user-facing changes?

Potentially this will affect how some queries are interpreted, where int literals greater than int64-max are provided.

### CODE REVIEW

**Comment by yukkit** [comment]
LGTM

**Comment by alamb** [comment]
THanks @jackkleeman  and @yukkit  -- I have kicked off the CI

File: datafusion/sql/src/expr/value.rs [hunk 1]
```diff
@@ -56,6 +56,8 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             ))
         } else if let Ok(n) = n.parse::<i64>() {
             Ok(lit(n))
+        } else if let Ok(n) = n.parse::<u64>() {
```
Comment (comphead) [inline datafusion/sql/src/expr/value.rs:59]: @jackkleeman please run `cargo fmt --all` and resubmit the PR

**Comment by alamb** [comment]
I took the liberty of fixing the `cargo fmt` issue and merging up from main

**Comment by alamb** [comment]
Thanks again @jackkleeman

**Pull request merged**